karate-core/src/main/java/com/intuit/karate/driver/DriverOptions.java

```diff
@@ -185,6 +186,9 @@ public DriverOptions(ScenarioContext context, Map<String, Object> options, LogAp
         highlight = get("highlight", false);
         highlightDuration = get("highlightDuration", Config.DEFAULT_HIGHLIGHT_DURATION);
         attach = get("attach", null);
+        emulateDevice = get("emulateDevice", new HashMap<String, Object>());
```

Comment:
also not keen on having DriverOptions have this Chrome specific stuff.

actually can you read the docs here: https://github.com/intuit/karate/tree/develop/examples/ui-test#devtools-protocol-tips

you should be able to start a test with url `about:blank` and then manually send this message

if that works, I would consider deferring adding `emulateDevice` because hardly anyone uses it. users can script any extra chrome behavior this way, another one that I expect to be used is simulating bandwidth speed / throttling